src/contract_wrappers/modules/transfer_manager/general_transfer_manager_wrapper.ts

```diff
@@ -457,6 +568,220 @@ export default class GeneralTransferManagerWrapper extends ModuleWrapper {
+      params.fromValidKYC,
+      params.toValidKYC,
+      params.fromRestricted,
+      params.toRestricted,
+      params.txData,
+      params.safetyFactor,
+    );
+    return result;
+  };
+
+  /**
+   * Modifies the successful checks required for transfers.
+   */
+  public modifyTransferRequirementsMulti = async (params: ModifyTransferRequirementsMulti) => {
+    assert.assert(await this.isCallerAllowed(params.txData, Perm.Admin), 'Caller is not allowed');
+    assert.assert(
+      params.transferTypes.length === params.fromValidKYC.length &&
+        params.fromValidKYC.length === params.toValidKYC.length &&
+        params.toValidKYC.length === params.fromRestricted.length &&
+        params.fromRestricted.length === params.toRestricted.length,
+      'Mismatched input lengths',
+    );
+    return (await this.contract).modifyTransferRequirementsMulti.sendTransactionAsync(
+      params.transferTypes,
+      params.fromValidKYC,
+      params.toValidKYC,
+      params.fromRestricted,
+      params.toRestricted,
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Add or remove KYC info of an investor.
+   */
+  public modifyKYCDataMulti = async (params: ModifyKYCDataMulti) => {
+    assert.assert(await this.isCallerAllowed(params.txData, Perm.Admin), 'Caller is not allowed');
+    assert.assert(
+      params.investors.length === params.canSendAfter.length &&
+        params.canSendAfter.length === params.canReceiveAfter.length &&
+        params.canReceiveAfter.length === params.expiryTime.length,
+      'Mismatched input lengths',
+    );
+    const canSendAfter: BigNumber[] = [];
+    const canReceiveAfter: BigNumber[] = [];
+    const expiryTime: BigNumber[] = [];
+
+    for (let i = 0; i < params.canSendAfter.length; i += 1) {
+      canSendAfter.push(dateToBigNumber(params.canSendAfter[i]));
+      canReceiveAfter.push(dateToBigNumber(params.canReceiveAfter[i]));
+      expiryTime.push(dateToBigNumber(params.expiryTime[i]));
+    }
+
+    return (await this.contract).modifyKYCDataMulti.sendTransactionAsync(
+      params.investors,
+      canSendAfter,
+      canReceiveAfter,
+      expiryTime,
+      params.txData,
+      params.safetyFactor,
+    );
+  };
+
+  /**
+   * Adds or removes addresses from the whitelist - can be called by anyone with a valid signature
+   */
+  public modifyKYCDataSignedMulti = async (params: ModifyKYCDataSignedMulti) => {
+    const canSendAfter: BigNumber[] = [];
+    const canReceiveAfter: BigNumber[] = [];
+    const expiryTime: BigNumber[] = [];
+    for (let i = 0; i < params.canSendAfter.length; i += 1) {
+      canSendAfter.push(dateToBigNumber(params.canSendAfter[i]));
+    }
+    for (let i = 0; i < params.canReceiveAfter.length; i += 1) {
+      canReceiveAfter.push(dateToBigNumber(params.canReceiveAfter[i]));
+    }
+    for (let i = 0; i < params.expiryTime.length; i += 1) {
+      expiryTime.push(dateToBigNumber(params.expiryTime[i]));
+    }
```

Comment:
All of the previous section can be rewritten as

const candSendAfter = params.canSendAfter.map(dateToBigNumber);
const canReceiveAfter = params.canReceiveAfter.map(dateToBigNumber);
const expiryTime = params.expiryTime.map(dateToBigNumber);
And you save yourself the need of a foor loop